### PR TITLE
feat: check and warning for installed incompatible add-ons

### PR DESCRIPTION
**Pull request by frankkopp**
## Summary of Changes
This PR adds a configurable list of incompatible add-ons to each installable add-on. 
If searches the configured community folder and reads each folder/manifest.json to compare to the incompatibility list. 
Incompatible add-ons are listed in a warning with the recommendation to remove them. 

- [x] Split configuration for Installation folder and Community folder (two separate configs now)
- [x] Incompatibility configuration structure
- [x] Implement search logic
- [x] Warning
- [x] CleanUp / Improve code

## Screenshots (if necessary)
Example data only! List of incompatible add-ons is defined in installer-data repo. 
image

## Testing Instructions
- Make sure to use this Installer config: https://cdn.flybywiresim.com/installer/config/pr-6.json (ask on Discord how to change)
- Configuration of Install folder and Community folder has been separated - test this as best possible
- See list of current known incompatible add-ons - install them an test Installer warning (sufficient to create a folder and place the add-on's manifest.json file in there).

Discord username (if different from GitHub): Cdr_Maverick#6475

### CODE REVIEW

**Comment by Revyn112** [comment]
The Headwind A330-900 mod is a standalone addon which doesn't overrider files of the a32nx, so this addon shouldn't be on the list.

**Comment by frankkopp** [comment]
> The Headwind A330-900 mod is a standalone addon which doesn't override files of the a32nx, so this addon shouldn't be on the list.

Not true. It overrides the build-info.json for example and might overwrite other files as well. 

As long as we have users reporting issues with the A32NX having Headwind installed, this warning is warrented.

**Comment by FoxtrotSierra6829** [comment]
We are not going to warn for an addon which is in our installer. Identify the issues and request to resolve them instead.

**Comment by frankkopp** [comment]
> 

Well - I agree for add-ons which ARE in our Installer. The Headwind isn't for good reason!

**Review by Benjozork — CHANGES_REQUESTED** [review]
Please make those changes without running a non-eslint formatter

File: src/renderer/utils/InstallManager.tsx [hunk 1]
```diff
@@ -256,7 +258,64 @@ export class InstallManager {
...
+        // Find incompatible add-ons
+        // This iterates through the first level of folders of the  MSFS Community folder looking for the manifest.json
+        // file. It compares the manifest.json file content with the configured incompatible add-ons (data.ts) and if it
+        // finds one or more matches, it will issue a warning.
+        console.log('Searching incompatible add-ons');
+        const manifestFileName = 'manifest.json';
+        const comDir = Directories.communityLocation();
+        const incompatibleAddons: AddonIncompatibleAddon[] = [];
+        try {
+            const addonFolders = fs.readdirSync(comDir);
```
Comment (Benjozork) [inline src/renderer/utils/InstallManager.tsx:270]: **suggestion:** extract this logic to some other file

File: src/renderer/utils/InstallManager.tsx [hunk 2]
```diff
@@ -256,7 +258,64 @@ export class InstallManager {
...
+                if (stat.isDirectory()) {
+                    const dirEntries = fs.readdirSync(filePath);
+                    if (dirEntries.includes(manifestFileName)) {
+                        const manifest = JSON.parse(fs.readFileSync(path.join(filePath, manifestFileName), 'utf8'));
+                        for (const item of addon.incompatibleAddons) {
+                            if ((!item.title || manifest.title === item.title) &&
+                                (!item.creator || manifest.creator === item.creator) &&
+                                (!item.package_version || manifest.package_version === item.package_version)
+                            ) {
+                                console.log("!!! %s: %s", manifest.title, item.description);
```
Comment (Benjozork) [inline src/renderer/utils/InstallManager.tsx:283]: **issue:** remove debug output

File: src/renderer/utils/InstallerConfiguration.ts [hunk 3]
```diff
@@ -152,6 +153,28 @@ export interface AddonDependency {
...
+ */
+export interface AddonIncompatibleAddon {
+    /**
+     * Fields from the addon's manifest.json
+     * This need to be configured identically to the addon's manifest.json
+     * Leaving a field empty ignores it for the search.
+     */
+    title?: string,
+    creator?: string,
+    package_version?: string,
```
Comment (Benjozork) [inline src/renderer/utils/InstallerConfiguration.ts:167]: **style:** config properties use camelCase, so `packageVersion`

File: src/renderer/utils/InstallManager.tsx [hunk 4]
```diff
@@ -256,7 +258,64 @@ export class InstallManager {
...
+            const addonFolders = fs.readdirSync(comDir);
+            for (const entry of addonFolders) {
+                const filePath = path.join(comDir, entry);
+                const stat = fs.statSync(filePath);
+                if (stat.isDirectory()) {
+                    const dirEntries = fs.readdirSync(filePath);
+                    if (dirEntries.includes(manifestFileName)) {
+                        const manifest = JSON.parse(fs.readFileSync(path.join(filePath, manifestFileName), 'utf8'));
+                        for (const item of addon.incompatibleAddons) {
+                            if ((!item.title || manifest.title === item.title) &&
```
Comment (Benjozork) [inline src/renderer/utils/InstallManager.tsx:279]: **style:** this if statement is very hard to read. I would split it into multiple conditions

**issue:** the version checking will fail any time the author updates their package, and will not allow detecting a range or set of versions. a package like `semver` should be used instead

**Review by Benjozork — CHANGES_REQUESTED** [review]

File: src/renderer/utils/InstallerConfiguration.ts (hunk 3 above)
Comment (frankkopp) [reply src/renderer/utils/InstallerConfiguration.ts:167]: ok will do - has been done on purpose as the manifest uses package_version - but either way is good for me

File: src/renderer/utils/InstallManager.tsx (hunk 1 above)
Comment (frankkopp) [reply src/renderer/utils/InstallManager.tsx:270]: That would be inconsistent - dependency check is also inline in the file.

File: src/renderer/utils/InstallManager.tsx (hunk 2 above)
Comment (frankkopp) [reply src/renderer/utils/InstallManager.tsx:283]: D idn 't consider this as debug output but as a valid log entry when inc. add-ons have been found. Format can be changed of course. '

File: src/renderer/utils/InstallManager.tsx (hunk 4 above)
Comment (frankkopp) [reply src/renderer/utils/InstallManager.tsx:279]: will change the style for better readability

An improved version check I wanted to do later in a separate PR. I like to iterate improvements after something was in production and we gathered some feedback and learnings.

File: src/renderer/utils/InstallManager.tsx (hunk 1 above)
Comment (Benjozork) [reply src/renderer/utils/InstallManager.tsx:272]: Nothing else here is doing complex filesystem work on things unrelated to the install itself. Hence why it has to go into another file. This is done the same way for external apps, background services, etc. and their related checks.

Comment (frankkopp) [reply src/renderer/utils/InstallManager.tsx:272]: Ok - I can extract this into a separate class if you think it is necessary. 
Will not be able to really work on this before Friday though.

**Pull request merged**